gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQueryError.java

```diff
@@ -0,0 +1,118 @@
+package com.google.gcloud.bigquery;
+
+import com.google.api.services.bigquery.model.ErrorProto;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Google Cloud BigQuery error.
+ */
+public class BigQueryError implements Serializable {
+
+  static final Function<ErrorProto, BigQueryError> FROM_PB_FUNCTION =
+      new Function<ErrorProto, BigQueryError>() {
+        @Override
+        public BigQueryError apply(ErrorProto pb) {
+          return BigQueryError.fromPb(pb);
+        }
+      };
+  static final Function<BigQueryError, ErrorProto> TO_PB_FUNCTION =
+      new Function<BigQueryError, ErrorProto>() {
+        @Override
+        public ErrorProto apply(BigQueryError error) {
+          return error.toPb();
+        }
+      };
+  private static final long serialVersionUID = -6566785320629096688L;
+
+  private final String reason;
+  private final String location;
+  private final String debugInfo;
+  private final String message;
+
+  BigQueryError(String reason, String location, String message, String debugInfo) {
+    this.reason = reason;
+    this.location = location;
+    this.debugInfo = debugInfo;
+    this.message = message;
+  }
+
+  BigQueryError(String reason, String location, String message) {
+    this.reason = reason;
+    this.location = location;
+    this.message = message;
+    this.debugInfo = null;
+  }
+
+  /**
+   * Returns short error code that summarizes the error.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/troubleshooting-errors">
+   *     Troubleshooting Errors</a>
+   */
+  public String reason() {
+    return reason;
+  }
+
+  /**
+   * Returns where the error occurred, if present.
+   */
+  public String location() {
+    return location;
+  }
+
+  String debugInfo() {
+    return debugInfo;
+  }
+
+  /**
+   * Returns a human-readable description of the error.
+   */
+  public String message() {
+    return message;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(reason, location, message);
+  }
+
+  @Override
+  public String toString() {
```

Comment:
Maybe unnecessary, but perhaps we could add tests for the `toString` method here and in the other classes.